src/extension/extension.ts

```diff
@@ -24,186 +44,205 @@ export function activate(context: vscode.ExtensionContext): void {
+    toDuplicate: ConfNameMap,
+    duplicated: ConfNameMap,
+  ): Promise<void> {
+    // get the content of the conf to duplicate
+    // cretate a new conf file with the name of "duplicated", content of "to duplicate", and open it with settings view
+    const path = vscode.workspace.workspaceFolders?.[0]
+    if (!path) return
+    const confFile = getConfFilePath(toDuplicate.fileName)
+    const confFileUri = vscode.Uri.joinPath(path.uri, confFile)
+    const decoder = new TextDecoder()
+    try {
+      const confFileContent: ConfFile = JSON.parse(
+        decoder.decode(await vscode.workspace.fs.readFile(confFileUri)),
+      )
+      confFileContent.msg = ''
+
+      try {
+        const newConfFilePath = getConfFilePath(duplicated.fileName)
+        const newConfFileUri = vscode.Uri.joinPath(path.uri, newConfFilePath)
+        const encoder = new TextEncoder()
+        const content = encoder.encode(JSON.stringify(confFileContent, null, 2))
+        await vscode.workspace.fs.writeFile(newConfFileUri, content)
+      } catch (e) {
+        vscode.window.showErrorMessage(`Can't create conf file. Error: ${e}`)
       }
-    })
+      SettingsPanel.setResultsWebviewProvider(resultsWebviewProvider)
+      SettingsPanel.render(context.extensionUri, duplicated, confFileContent)
+    } catch (e) {
+      vscode.window.showErrorMessage(
+        `Can't read conf file: ${confFile}. Error: ${e}`,
+      )
+    }
   }
 
-  async function runScript() {
-    quickPickWithConfFiles((selection, quickPick) => {
-      if (selection[0]) {
-        const confFile = selection[0].label
-
-        if (confFile.includes(' ')) {
-          vscode.window.showErrorMessage(
-            'The extension does not support path to conf file with spaces. Correct the conf file path and run the script again',
-          )
-          quickPick.hide()
-          return
-        }
+  function getConfFilePath(name: string): string {
+    return 'conf/' + name + '.conf'
+  }
 
-        scriptRunner.run(confFile)
-        quickPick.hide()
-      }
-    })
+  async function runScript(name: ConfNameMap) {
+    SettingsPanel.removePanel(name.displayName)
+    const confFile = getConfFilePath(name.fileName)
+    scriptRunner.run(confFile)
+  }
+
+  function getConfUri(name: string): vscode.Uri | void {
+    const path = vscode.workspace.workspaceFolders?.[0]
+    if (!path) return
+    const confFile = getConfFilePath(name)
+    const confFileUri = vscode.Uri.joinPath(path.uri, confFile)
+    return confFileUri
+  }
+
+  function deleteConfFile(name: ConfNameMap): void {
+    const confFileUri: vscode.Uri | void = getConfUri(name.fileName)
+    if (confFileUri) {
+      vscode.workspace.fs.delete(confFileUri)
+    }
+    SettingsPanel.removePanel(name.displayName)
+    scriptRunner.removeRunningScriptByName(name.fileName)
   }
 
   const resultsWebviewProvider = new ResultsWebviewProvider(
     context.extensionUri,
-    runScript,
-    showSettings,
   )
+
```

Comment:
try changing order to remove removeRunningScriptByName()

Reply:
This is for the results webview provider.
ScriptRunner object has to recieve ResultsWebviewProvider object when it is initialized
But, ResultsWebviewProvider has to recieve resultsWebviewProvider.removeScript = removeRunningScriptByName which is a ScriptRunner method.
The solution to the circle is this function, that is passed to ResultsWebviewProvider after ScriptRunner is initialized.

**relevant documentation was added above the function**